app/Jobs/PollForMediaWikiJobsJob.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Wiki;
+use Illuminate\Support\Facades\Http;
+
+class PollForMediaWikiJobsJob extends Job
+{
+    public function handle (): void
+    {
+        $wikis = Wiki::all()->pluck('domain');
+        foreach ($wikis as $wikiDomain) {
+            if ($this->hasPendingJobs($wikiDomain)) {
+                $this->enqueueWiki($wikiDomain);
+            }
+        }
+    }
+
+    private function hasPendingJobs (string $wikiDomain): bool
+    {
+        $response = Http::withHeaders([
+            'host' => $wikiDomain
+        ])->get(
+            getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=query&meta=siteinfo&siprop=statistics&format=json'
+        );
+
+        if ($response->failed()) {
```

Comment:
Not sure about the error handling here. If one wiki breaks for whatever reason, it would block other wikis. Is there an idiomatic way of returning an error so the iterating loop can collect these, and fail the job after as many wikis as possible have been processed?

Reply:
E.g. instead of returning `bool` from `hasPendingJobs` I could return `bool|SomeCustomException` and then type switch on the result, but I have no idea if this is considered good practice, or if there are better alternatives?

Reply:
Maybe it would be enough to log and mark the job as failed here instead?